app/models/research_session.rb

```diff
@@ -47,4 +53,12 @@ class ResearchSession < ApplicationRecord
   def reached_step?(step)
     Steps.instance.reached_step?(self, step)
   end
+
+  def set_slug_from_name!
+    return if name.blank?
+
+    slug = name.strip.downcase.tr(' ', '-')
```

Comment:
What if people use punctuation such as "My research project for Barnardo's!"?

Reply:
Or even an emoji "My research project for Barnardo's! 🤘"

Reply:
![image](https://user-images.githubusercontent.com/893208/32166662-79e44d72-bd5e-11e7-998c-001c624502a0.png)

Doesn't seem to be a problem!